### PR TITLE
Fixes for CI benchmarks: iteration 2

### PR DESCRIPTION
**Pull request by filipecosta90**
- Including module artifact check on benchmarks. 
- Included extra dev docs explaining benchmark triggering.
- Increases the JSON.ARRAPPEND test duration given the major regression ( fixed by #324 ) is now merged.
- Fixes some wrong benchmark names 
- Make usage of readies/bin/getterraform and remove the explicit code for fetching terraform from repo

### CODE REVIEW

**Comment by rafie** [comment]
@filipecosta90 I fixed the rust installation issue. However, I'm wary of the build cache (esp. how it's used here). I think we should stop using it. WDYT?

File: .circleci/config.yml [hunk 1]
```diff
@@ -364,7 +365,7 @@ workflows:
           requires:
             - deploy_release
       - performance-automation:
-          <<: *on-integ-and-version-tags
+          <<: *not-on-integ-branch
```
Comment (rafie) [inline .circleci/config.yml:368]: shouldn't that be `on-integ-and-version-tags`?

Comment (filipecosta90) [reply .circleci/config.yml:368]: you're absolutly right @rafie . addressing it

Comment (filipecosta90) [reply .circleci/config.yml]: @rafie fixed

**Review by rafie — APPROVED** [review]

**Pull request merged**